app/models/concerns/subform_utilities.rb

```diff
@@ -0,0 +1,55 @@
+module SubformUtilities
+  extend ActiveSupport::Concern
+
+  included do
+    extend(ClassMethods)
+  end
+
+  module ClassMethods
+    def distinct_by_column(name)
+      select("distinct on (#{self.name.underscore.pluralize}.#{name}) #{self.name.underscore.pluralize}.*")
```

Comment:
this is an SQL injection issue because `name` is unparsed user input.
best option I think is to whitelist the incoming `name` param and ensure it belongs to a certain list of acceptable fields.
`(x, y z).include?(name)`

Reply:
@steveyken , here the records are getting filtered by column name. Only distinct records will be sent in response for brand. It is not user input. While requesting from client, it is asking for distinct by "brand" records.

``` let details = await this.store.query(selectedSubform, {
              distinct: "brand"
            });```

That's why whitelisting it not required. 

Please let me know your thoughts on this.

Reply:
hi @abhinav098 by 'user input' I mean that the value is coming directly from the params hash which is being used to set name here. It's not safe just because you are setting 'brand' in the end user app. Consider the case where a hacker sends API requests direct to the system bypassing the app. They can put whatever they like in the `params[:distinct]` field. You could even change this in the Chrome debugger if you wanted.

Use `ActiveRecord.santize_sql` and lookup the column name to be safe. Here's an untested example to help get started:

```
def distinct_by_column(column)
  raise StandardError('Column name invalid') unless self.column_names.include?(col)
  tbl_name = self.name.tableize # self.name.underscore.pluralize
  sql_select = self.send(:sanitize_sql, ['distinct on (?.? ?.*)', tbl_name, column, tbl_name])
  select(sql_select)
end
```

Thanks!

Reply:
Let me know if I haven't explained it well enough - it's important we're on the same page on this one!